autotest/OldTests/src/test/java/com/tle/webtests/test/users/LDAPTest.java

```
@@ -0,0 +1,198 @@
+    assertTrue(userDialog.search("william").containsUsername("william.bowling"));
+
+    logout(context);
+  }
+  // acls, itemdefinition, properties
+  @Test
+  public void testViaAdvancedScripts() {
+    // TODO: open ldap and roles(?)
+    final List<String> charliesGroups = Arrays.asList("dev", "support");
+    PageContext context = newContext("ldapad");
+    logon(context, LDAP_USER_NICK, LDAP_PASSWORD);
+    WizardPageTab openWizard = new MenuSection(context).get().clickContribute("Basic Items");
+    openWizard.editbox(1, getClass().getSimpleName() + " testing");
+    SummaryPage summary = openWizard.save().publish();
+    List<String> values = summary.getValuesByCustomDisplay();
+    // search groups for "alpha" group
+    assertEquals(values.get(0), "dev");
+    // members of alpha and it's child groups
+    assertEqualsNoOrder(
+        values.get(1).split(", "),
+        new String[] {"[Andrew Gibb [andrew.gibb]", "Nick Charles [nick.charles]]"});
+    // first group returned that nick is a member of
+    assertTrue(charliesGroups.contains(values.get(2)));
+    // is nick a member of this group (always)
+    assertTrue(Boolean.valueOf(values.get(3)));
+    // what groups is nick a member of
+    assertEqualsNoOrder(values.get(4).split(", "), charliesGroups.toArray());
+    logout();
+  }
+
+  private void startLdap() {
+    try {
+      InMemoryDirectoryServerConfig config = new InMemoryDirectoryServerConfig("dc=edalex,dc=com");
+      config.addAdditionalBindCredentials("cn=equella", "password");
+      config.setBaseDNs("dc=edalex,dc=com");
+      config.setSchema(null);
+      config.setListenerConfigs(InMemoryListenerConfig.createLDAPConfig("default", 10389));
+      ldapServer = new InMemoryDirectoryServer(config);
+      ldapServer.importFromLDIF(true, Attachments.get("ldap_test.ldif").getPath());
+      ldapServer.startListening();
+    } catch (LDAPException e) {
+      e.printStackTrace();
```

Should throw a runtime exception